src/core/namespace.c

```diff
@@ -940,13 +940,12 @@ static int mount_procfs(const MountEntry *m, const NamespaceInfo *ns_info) {
 }
```

Comment:
i don't grok this:

"But if a heap buffer is used, suddenly it fails."

This sounds like a memory corruption. Why would it fail if the memory is allocated differently? This needs investigation...

Reply:
See other comment - this works only if the MountEntry is set up using path_const, as it's side-stepping the accessor:

inner_path = m->path_const;

If a helper function is used, as I was doing in another PR, then mount_tmpfs explodes since path_const is never set.